sections/main-product.liquid

```diff
@@ -24,15 +24,16 @@
           {{ "accessibility.skip_to_product_info" | t }}
         </a>
         <ul class="product__media-list grid grid--peek list-unstyled slider slider--mobile" role="list">
+          {%- assign variant_images = product.images | where: 'attached_to_variant?', true | map: 'src' -%}
           {%- if product.selected_or_first_available_variant.featured_media != null -%}
             {%- assign media = product.selected_or_first_available_variant.featured_media -%}
-            <li class="product__media-item grid__item slider__slide{% if media.media_type != 'image' %} product__media-item--full{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
+            <li class="product__media-item grid__item slider__slide{% if media.media_type != 'image' %} product__media-item--full{% endif %}{% if variant_images contains media.src and section.settings.hide_variants %} product__media-item--variant{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
               {% render 'product-thumbnail', media: media, position: 'featured', loop: section.settings.enable_video_looping, modal_id: section.id %}
             </li>
           {%- endif -%}
           {%- for media in product.media -%}
             {%- unless media.id == product.selected_or_first_available_variant.featured_media.id -%}
-              <li class="product__media-item grid__item slider__slide{% if media.media_type != 'image' %} product__media-item--full{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
+              <li class="product__media-item grid__item slider__slide{% if media.media_type != 'image' %} product__media-item--full{% endif %}{% if variant_images contains media.src and section.settings.hide_variants %} product__media-item--variant{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
```

Comment:
Is it the expectation that these images are visible INSIDE the product modal? I find the setting naming a little confusing if that's the case - I would expect it to hide everywhere.

https://user-images.githubusercontent.com/37168033/125852374-a46b13c8-c046-4005-9f77-240ae8c52756.mp4

Right now I can see all the variant images when I open the modal.

Reply:
Same here.

Reply:
I wonder 🤔 @melissaperreault how do you think it should work in the case mentioned above ^ Should we also hide them in the modal ?

Reply:
The expectation is to hide everywhere when the setting is true, I believe it was the case when I tested if I recall. Otherwise it was overlooked, when this setting is enabled, it applies to all devices! 🙌

Reply:
Made some changes in this commit to deal with it: https://github.com/Shopify/dawn/pull/158/commits/263b8919929565718b5df33359138911e6b5b831